File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int maxProfit=0;
        int min=Integer.MAX_VALUE;
        for(int i=0;i<n;i++){
            if(prices[i]<min){
                min= prices[i];
            }else{
                int profit=prices[i]-min;
                maxProfit=Math.max(profit,maxProfit);
            }
        }
        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        maxprofit = 0
        min = prices[0]
        for i in range(0,n):
            if prices[i]<min:
                min = prices[i]
            else:
                profit = prices[i]-min
                maxprofit=max(maxprofit,profit)
        return maxprofit